finetuner/tuner/keras/__init__.py

```diff
@@ -184,6 +187,9 @@ def fit(
 
                 self._trigger_callbacks('on_epoch_end')
 
+                if self.stop_training:
```

Comment:
@gvondulong I don't think we need to return the best model - we have the Checkpoint callback for saving. The checkpoint should then also facilitate loading the best model. The `Tuner`'s fit function is not supposed to return anything. Otherwise there is just too much coupling